src/plugins/navigation_embeddable/public/embeddable/navigation_embeddable_factory.ts

```diff
@@ -33,9 +33,19 @@ export class NavigationEmbeddableFactoryDefinition
   implements EmbeddableFactoryDefinition<NavigationEmbeddableInput>
 {
   public readonly type = NAVIGATION_EMBEDDABLE_TYPE;
-
   public isContainerType = false;
 
+  private openEditorFlyout?: (
+    initialInput?: Omit<NavigationEmbeddableInput, 'id'>,
+    parentDashboard?: DashboardContainer
+  ) => Promise<Partial<NavigationEmbeddableInput>>;
+
+  constructor() {
+    import('../editor/open_editor_flyout').then(({ openEditorFlyout }) => {
+      this.openEditorFlyout = openEditorFlyout;
+    });
+  }
```

Comment:
Moving the async import to the constructor allows the flyout to open **instantly**, even when the network connection is slow:

**Before:**

**After:**

Reply:
Ahhh, that's a good catch... I was just using the `Slow 3G` preset in the "before" video 🤔 

> I wonder if it makes sense to open an empty flyout and asynchronously load the component?
Hrmmm I could try that... When I tried to synchronously import the flyout in the factory, the `page load bundle size` jumped up like CRAZZYYY - but that's probably because everything inside of the flyout is also imported synchronously.

If we imported the flyout synchronously in the factory, but made all of the flyout's imports async.... Maybe that would solve things.

Reply:
Alright, I managed to... **half** improve it with c1daec28cf0ea8066e73c5adb2a0ea42740b10c9? 

https://github.com/elastic/kibana/assets/8698078/1db68ce4-7023-424e-bc46-0691d4a0be17

Since we are back to async importing `openEditorFlyout` in `getExplicitInput`, there is still a delay on the first click.... But the contents of the flyout are now also async imported, so that delay is at least shortened a bit. Not sure if this is better? 🤔

Reply:
@Heenawter this isn't _so_ bad, so if there isn't an easy way to split it up further I'm okay with this solution.

I wonder though, how complex is the `openEditorFlyout` function that actually opens the flyout? Is there any way to make that function _very_ barebones so that it can go in the page load bundle and open immediately so that the flyout with the title `Create Links panel` and the spinner show right away?

That would be the ideal solution IMO.

Reply:
Looked into this, and `openEditorFlyout` is (I believe) as bare bones as it can be... Right now, if I synchronously import `openEditorFlyout` in the factory and add it to the page load bundle, it adds`~18kB` to the size. The bulk of this seems to be coming from the `withSuspense` import - specifically, it adds over `10kB`... And I don't think there's a way to separate that out? 🤔 I tried, and just ended up making things worse, hahah :)